carbon-design-system/carbon-addons-iot-react
# 3044 menu button style audit button variations

Closes #3044

**Summary**
Fixed the styling remarks from #3044 and also fixed some additional RTL issued. In order to understand the changes it is easiest to go through the commits one by one.

**Change List (commits, features, bugs, etc)**

1. Fixed button and menu item sizes according to the spec  + unit tests
2. Added button variations (kind) according to the spec + unit tests
3. Fixed the shadow from the menu that was incorrectly dropping in the button (complex fix needed) + unit tests
4. Fixed additional RTL issues


**Acceptance Test (how to verify the PR)**
Start by reading the spec.

1. Go to https://deploy-preview-3185--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-menubutton--single-menu-button and use the size knob to verify that the sizes renders according to spec. Note, we use "default" instead of large until carbon v.11

2.  Repeat step 1 for https://deploy-preview-3185--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-menubutton--split-menu-button and https://deploy-preview-3185--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-menubutton--icon-only-menu-button

3. Verify that https://deploy-preview-3185--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-menubutton--single-menu-button renders the different kinds of buttons available in the kind knob and that the button and the menu looks as expected. Compare to the spec.

4. Verify that https://deploy-preview-3185--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-menubutton--split-menu-button renders the different kinds of buttons available in the kind knob and that the button and the menu looks as expected. Compare to the spec.

5.  Go to https://deploy-preview-3185--carbon-addons-iot-react.netlify.app?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-menubutton--auto-positioning-example and for all COMBINATIONS of positions, sizes and RTL-mode for the IconOnlyButton verify that: 
a) the shadow from the menu does not fall on the button (for any state of interaction)
b) the blue focus border of the menu is intact
c) the blue focus border of the button is intact (to test this click the button to open the menu and then press down on the button with the mouse again but move the cursor outside the button before releasing)


**Regression Test (how to make sure this PR doesn't break old functionality)**

- All stories work as before except for the changes described by the spec.



**Things to look for during review**
